ui/src/app/settings/containers/Storage/Storage.js

```diff
@@ -1,5 +1,125 @@
+import { Formik, Form } from "formik";
 import React from "react";
+import { useSelector } from "react-redux";
 
-const Storage = () => <h4>Storage</h4>;
+import selectors from "app/settings/selectors";
+import Button from "app/base/components/Button";
+import Input from "app/base/components/Input";
+import Loader from "app/base/components/Loader";
+import Select from "app/base/components/Select";
+
+const getStorageLayoutWarning = layout => {
+  switch (layout) {
+    case "blank":
+      return "You will not be able to deploy machines with this storage layout. Manual configuration is required.";
+    case "vmfs6":
+      return "The VMFS6 storage layout only allows for the deployment of VMware (ESXi).";
+    default:
+      return;
+  }
+};
+
+const Storage = () => {
+  const loaded = useSelector(selectors.config.loaded);
+
+  const defaultStorageLayout = useSelector(
+    selectors.config.defaultStorageLayout
+  );
+  const storageLayoutOptions = useSelector(
+    selectors.config.storageLayoutOptions
+  );
+  const enableDiskErasing = useSelector(selectors.config.enableDiskErasing);
+  const diskEraseWithSecure = useSelector(selectors.config.diskEraseWithSecure);
+  const diskEraseWithQuick = useSelector(selectors.config.diskEraseWithQuick);
+
+  return (
+    <>
+      <h4>
+        Storage
+        {!loaded && <Loader text="Loading..." inline />}
```

Comment:
Should thing just be `loading`, in case the logic changes at some point which means the component loads as the result of an event etc?